### PR TITLE
[FIX] sale_timesheet: set the allocated hours correctly on project

### PR DESCRIPTION
**Pull request by mba-odoo**
- Steps:
 Create 4 Services:
	> Product Type : Service
	> Create on Order : Project & Task
	> Unit of Measure -respectively- :
		>> Hours (H),
		>> Days  (D),
		>> Units (U),
		>> g     (Other).
 Create a SO with them (1 uom_qty for each) and confirm.

- Expected:
 For the 'Allocated Hours' on the project,
 the conversion to time should be processed as follows :
H     : qty =     uom_qty [Hours]
D     : qty = uom_qty * 8 [Hours]
U     : qty =     uom_qty [Hours]
Other : qty =           0 [Hours] or uom_qty [uom]

Before this commit, it was going to convert the other unit
(like: g,kg,dozen,etc..) to the time unit which is not expected

So in this commit fix corrects everything as expected.

task-2858529

### CODE REVIEW

File: addons/sale_timesheet/models/project.py [hunk 1]
```diff
@@ -197,19 +197,23 @@ def _compute_allocated_hours(self):
...
-            ], 0.0)
-            # Now convert to the unit of measure to hours
-            allocated_hours *= uom_hour.factor
+            allocated_hours = 0.0
+            for product_uom, product_uom_qty in sol_qty_dict[project_id]:
+                if uoms_dict.get(product_uom) or product_uom == uom_unit.id:
+                    allocated_hours += product_uom_qty * uoms_dict.get(product_uom, project.timesheet_encode_uom_id).factor_inv
+
+                    # Now convert to the unit of measure to hours
+                    allocated_hours *= uom_hour.factor
```
Comment (auon-odoo) [inline addons/sale_timesheet/models/project.py:216]: This is incorrect. With your commit message's usecase, I end up with allocated = 129.0
Maybe this :
```suggestion
                    allocated_hours += product_uom_qty * uoms_dict.get(product_uom, project.timesheet_encode_uom_id).factor_inv * uom_hour.factor
```
? Then I end up with 10.0

**Review by auon-odoo — COMMENTED** [review]
Thanks for your work :+1: 
I left some comments

File: addons/sale_timesheet/models/project.py (hunk 1 above)
Comment (mba-odoo) [reply addons/sale_timesheet/models/project.py]: @auon-odoo , Done all the changes. Thanks

**Review by auon-odoo — APPROVED** [review]
Looks good, Thanks :)

**Review by xavierbol — COMMENTED** [review]
Thanks for the fix, we have another fix done in #99785. Normally we will also do the fix in 15.3. So I think, we could merge the both PR to first remove the compute, and apply your fix here:

<https://github.com/odoo/odoo/pull/99785/files#diff-fe32ed10bb425741f6bb32e82b6462195317fa6fd32a24d3eb9469185c63bd1aR237-R248>

As you prefer, we could wait the merge the fix in #99785.
It is up to you, let me know what you prefer. 🙂

**Comment by mba-odoo** [comment]
@xavierbol , as per my discussion with @ppr-odoo, will cherry-pick the commit in his branch and update the commit accordingly. So I am going to close this one. 
Thanks

**Comment by xavierbol** [comment]
Great thanks 😊

**Pull request closed without merging**